napari/plugins/_builtins.py

```diff
@@ -27,3 +35,144 @@ def napari_get_reader(path: Union[str, List[str]]) -> ReaderFunction:
         function that returns layer_data to be handed to viewer._add_layer_data
     """
     return image_reader_to_layerdata_reader(magic_imread)
+
+
+@napari_hook_implementation(trylast=True)
+def napari_write_image(path: str, data: Any, meta: dict) -> bool:
+    """Our internal fallback image writer at the end of the plugin chain.
+
+    Parameters
+    ----------
+    path : str
+        Path to file, directory, or resource (like a URL).
+    data : array or list of array
+        Image data. Can be N dimensional. If meta['rgb'] is `True` then the
+        data should be interpreted as RGB or RGBA. If meta['is_pyramid'] is
+        True, then the data should be interpreted as an image pyramid.
+    meta : dict
+        Image metadata.
+
+    Returns
+    -------
+    bool : Return True if data is successfully written.
+    """
+    ext = os.path.splitext(path)[1]
+    path += '.tif' if not ext else ''
+    imsave(path, data)
+    return True
+
+
+@napari_hook_implementation(trylast=True)
+def napari_write_points(path: str, data: Any, meta: dict) -> bool:
+    """Our internal fallback points writer at the end of the plugin chain.
+
+    Append `.csv` extension to the filename if it is not already there.
+
+    Parameters
+    ----------
+    path : str
+        Path to file, directory, or resource (like a URL).
+    data : array (N, D)
+        Coordinates for N points in D dimensions.
+    meta : dict
+        Points metadata.
+
+    Returns
+    -------
+    bool : Return True if data is successfully written.
+    """
+    ext = os.path.splitext(path)[1]
+    if ext == '':
+        path = path + '.csv'
+    elif ext != '.csv':
+        raise ValueError('If an extension is provided then it must be `.csv`')
```

Comment:
@tlambert03 should this `return False` instead of raising the ValueError? Should we emit a warning? Curious what you see as best practices that we should embody

Reply:
yeah, I think that should be the API (until we decide otherwise), changed it.
my reasoning is that if plugins _know_ they can't handle the request, they should gracefully exit without us having to handle the exceptions.  Let's leave exceptions for actual errors.